probe-rs/src/architecture/arm/communication_interface.rs

```diff
@@ -69,6 +69,16 @@ impl RegisterParseError {
 
 /// To be implemented by debug probe drivers that support debugging ARM cores.
 pub trait ArmProbeInterface: DapAccess + SwdSequence + SwoAccess + Send {
+    /// Reinitialize the communication interface (in place).
+    ///
+    /// Some chip-specific reset sequences may disable the debug port. `reinitialize` allows
+    /// a debug sequence to re-initialize the debug port, staying true to the `Initialized`
+    /// type state.
+    ///
+    fn reinitialize(&mut self) -> Result<(), ArmError> {
+        Ok(())
```

Comment:
I'd prefer this to not have a default implementation. 

I think it can be a bit confusing if this gets used, but it doesn't actually do anything. If this isn't necessary to be implemented for a probe, then the implementation of the trait should have to decide on that, and explain why reinitializing the connection isn't necessary.

Reply:
Alright, I've added empty traits to each implementor and removed the default trait.